apiserver/facades/agent/uniter/storage.go

```diff
@@ -406,3 +415,54 @@ func accessUnitTag(tag string, canAccess func(names.Tag) bool) (names.UnitTag, e
 	}
 	return u, nil
 }
+
+// watchStorageAttachment returns a state.NotifyWatcher that reacts to changes
+// to the VolumeAttachmentInfo or FilesystemAttachmentInfo corresponding to the
+// tags specified.
+func watchStorageAttachment(
+	st storageInterface,
+	stVolume storageVolumeInterface,
+	stFile storageFilesystemInterface,
+	storageTag names.StorageTag,
+	machineTag names.MachineTag,
+	unitTag names.UnitTag,
+) (state.NotifyWatcher, error) {
+	storageInstance, err := st.StorageInstance(storageTag)
+	if err != nil {
+		return nil, errors.Annotate(err, "getting storage instance")
+	}
+	var watchers []state.NotifyWatcher
+	switch storageInstance.Kind() {
+	case state.StorageKindBlock:
+		volume, err := stVolume.StorageInstanceVolume(storageTag)
```

Comment:
Don't you need to check for nil stVolume here?

Reply:
done